libs/sdk-ui-pivot/src/impl/structure/headers/ColumnHeader.tsx

```diff
@@ -78,14 +101,37 @@ const ColumnHeader: React.FC<IColumnHeaderProps> = (props) => {
         ? (isSliceMeasureCol(col) && displayName) || isMixedHeadersCol(col)
         : true;
 
+    const [isFocused, setIsFocused] = useState(false);
+
+    useEffect(() => {
+        if (!api) {
+            return;
+        }
+        const handleFocusChange = (event: HeaderFocusedEvent) => {
+            const focusedColId = event.column.isColumn ? event.column.getColId() : event.column.getGroupId();
+            const myColId = column.getColId();
+            setIsFocused(focusedColId === myColId);
+        };
+        const handleCellFocused = () => {
+            setIsFocused(false);
+        };
+        api.addEventListener("headerFocused", handleFocusChange);
+        api.addEventListener("cellFocused", handleCellFocused);
+
+        return () => {
+            api.removeEventListener("headerFocused", handleFocusChange);
+            api.removeEventListener("cellFocused", handleCellFocused);
+        };
+    }, [api, column]);
+
     return (
         <HeaderCell
             className={cx("s-pivot-table-column-header", className)}
             textAlign={textAlign}
             displayText={displayName}
             enableSorting={isSortingEnabled}
             sortDirection={sorting}
-            defaultSortDirection={getDefaultSortDirection()}
+            defaultSortDirection={column.getColDef().sortingOrder?.[0] ?? undefined}
```

Comment:
`??` branch is called only when the statement before it evaluated as `undefined`. Isn't explicit `undefined` redundant?

Reply:
My bad, it looks like it also runs the other branch for `null`, hence its name `nullish coalescing operator` 🤦